app.rb

```diff
@@ -0,0 +1,80 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require './config/data_mapper'
+require 'pry'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  enable :method_override
+
+  get '/' do
+    @messages = Message.all
+    erb :index
+  end
+
+  get '/signup' do
+    erb :signup
+  end
+
+  post '/signup' do
+    user = User.create(name: params[:name], username: params[:username], email: params[:email], password: params[:password])
+    session[:id] = user.id
+    redirect '/signin'
+  end
+
+  get '/signin' do
+    erb :signin
+  end
+
+  post '/signin' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:id] = user.id
+      redirect("/profile/#{session[:id]}")
+    else
+      redirect '/'
+    end
+  end
+
+  get '/profile/:id' do
+    if signed_in?
+      @user = User.get(params[:id])
+      @users = User.all
```

Comment:
As we said, try leaving this out and seeing if the app still works as it doesn't seems @users is being used anywhere.